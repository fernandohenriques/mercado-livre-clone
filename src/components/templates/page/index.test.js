import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';

import Page from './index';

describe('<Page />', () => {
  it('Snapshot testing', () => {
    const { asFragment } = render(
      <Page title="Busca de Produtos">
        <h1>Busca</h1>
      </Page>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render page with document title right', () => {
    const title = 'Performes no Mercado Livre';
    const wrapper = shallow(
      <Page title={title}>
        <h1>Resultado da Busca</h1>
      </Page>
    );

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.find('Head')).toHaveLength(1);
    expect(wrapper.find('title')).toHaveLength(1);
    expect(wrapper.find('title').text()).toEqual(title);
  });

  it('Should render page with semantic tag main', () => {
    const wrapper = shallow(
      <Page title="Página de produto">
        <h1>Página de de um produto</h1>
      </Page>
    );

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.find('main')).toHaveLength(1);
  });
});
