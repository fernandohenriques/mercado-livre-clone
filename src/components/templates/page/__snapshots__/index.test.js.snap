// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> Snapshot testing 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <figure>
        <img
          alt="Mercado Livre - Onde comprar e vender de Tudo"
          src="logo.png"
          title="Mercado Livre - Onde comprar e vender de Tudo"
        />
      </figure>
      <div
        class="container"
      >
        <input
          class="input"
          placeholder="Buscar produtos..."
          type="text"
        />
        <button>
          <img
            alt="Buscar"
            src="ic-search.png"
            title="Buscar"
          />
        </button>
      </div>
    </div>
  </header>
  <main>
    <h1>
      Busca
    </h1>
  </main>
</DocumentFragment>
`;
